mozdownload/scraper.py

```diff
@@ -332,10 +335,15 @@ def show_matching_builds(self, builds):
 class DailyScraper(Scraper):
     """Class to download a daily build from the Mozilla server."""
 
-    def __init__(self, branch='mozilla-central', build_id=None, date=None,
+    def __init__(self, branch=None, build_id=None, date=None,
                  build_number=None, revision=None, *args, **kwargs):
         """Create an instance of the daily scraper."""
-        self.branch = branch
+        if not branch:
+            application = kwargs.get('application', None)
```

Comment:
Sorry for being late with my reply too! I somehow completely missed it in my inbox. :(

So the failure occurs because `build_index` will now only be available when you have called download. Maybe in those cases when no `download` is wanted, we should offer a method like `fetch` which would build the appropriate properties, and should be called before accessing any of those.